### PR TITLE
Fix invite notifications for users without pushers

### LINKED ISSUE #12839: Invites push actions don't get persisted and can't be queried using /notifications if the user doesn't have any pusher registered
### Description

For invites there exists a special push rule: `.m.rule.invite_for_me`. This rule seems to never get evaluated and added to the event_push_actions table, which then causes /notifications to never return pending invites as notifications.

### Steps to reproduce

- Invite another user
- Search for the invite using the following query: `SELECT * FROM event_push_actions WHERE event_id = '$event'`

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

I would like invites to actually show up as notifications!

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- **Homeserver**: neko.dev

If not matrix.org:

<!--
 What version of Synapse is running?

You can find the Synapse version with this command:

$ curl http://localhost:8008/_synapse/admin/v1/server_version

(You may need to replace `localhost:8008` if Synapse is not configured to
listen on that port.)
-->
- **Version**: 1.59.1

- **Install method**: ebuild
<!-- examples: package manager/git clone/pip  -->

- **Platform**: Gentoo
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->

### PR DESCRIPTION
**Pull request by deepbluev7**
fixes #12839
introduced in #705

Signed-off-by: Nicolas Werner <nicolas.werner@hotmail.de>

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by deepbluev7** [comment]
If anyone remembers, why that condition was added in 2016, I would be open to hearing it. But it seems to just arbitrarily break the /notifications endpoint, if you don't register any pusher.

File: changelog.d/12840.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix an issue introduced in Synapse 0.34 where /notifications would only return notifications if a user registered at least one pusher. Contributed by Famedly.
```
Comment (babolivier) [inline changelog.d/12840.bugfix:1]:
```suggestion
Fix an issue introduced in Synapse 0.34 where the `/notifications` endpoint would only return notifications if a user registered at least one pusher. Contributed by Famedly.
```

**Review by babolivier — CHANGES_REQUESTED** [review]
Thanks for the contribution! Could you add a test for it to ensure we don't accidentally break this in the future please?

**Comment by babolivier** [comment]
For the paper trail, this bug was introduced in https://github.com/matrix-org/synapse/commit/ed3979df5faac6d63990f4230662ff8cdcf59584

**Comment by deepbluev7** [comment]
Sure, let me try to add a test for that. Originally I wanted to add a sytest test, but... that looked like a lot of effort to synchronize the branches. I guess a unit test should work too.

**Comment by deepbluev7** [comment]
Added a testcase now. (Also verified that it fails without this change)

**Comment by deepbluev7** [comment]
The complement failure is a network error...

**Review by babolivier — APPROVED** [review]

**Pull request merged**